GeometryService/src/STMMaker.cc

```diff
@@ -193,22 +203,84 @@ namespace mu2e {
                            _SSCollimatorHole1xOffset,    
                            _SSCollimatorHole1RadiusUpStr,
                            _SSCollimatorHole1RadiusDnStr,
-                           _SSCollimatorHole1LinerBuild,
-                           _SSCollimatorHole1LinerThickness,
+                           _SSCollimatorHole1Liner1Build,//wyq
+                           _SSCollimatorHole1Liner1Thickness,//wyq
+                           _SSCollimatorHole1Liner2Build,//wyq
+                           _SSCollimatorHole1Liner2Thickness,//wyq
                            _SSCollimatorHole2Build,      
                            _SSCollimatorHole2xOffset,    
                            _SSCollimatorHole2RadiusUpStr,
                            _SSCollimatorHole2RadiusDnStr,
-                           _SSCollimatorHole2LinerBuild,
-                           _SSCollimatorHole2LinerThickness,
+                           _SSCollimatorHole2Liner1Build,//wyq
+                           _SSCollimatorHole2Liner1Thickness,//wyq
+                           _SSCollimatorHole2Liner2Build,//wyq
+                           _SSCollimatorHole2Liner2Thickness,//wyq
                            _SSCollimatorOffsetInMu2e,
                            _SSCollimatorRotation,
                            _SSCollimatorMaterial,        
                            _SSCollimatorLinerMaterial,
-                           _SSCollimatorHoleLinerMaterial
+                           _SSCollimatorHole1Liner1Material,//wyq
+                           _SSCollimatorHole1Liner2Material,//wyq
+                           _SSCollimatorHole2Liner1Material,//wyq
+                           _SSCollimatorHole2Liner2Material//wyq
                           ));       
     //}
+    ////wyq Shielding
+			if(_ShieldingHalfLength==0){
+				_ShieldingHalfLength=(_stmZAllowed-_SSCollimatorHalfLength*2)/2.0;
+			}
+			if(_ShieldingHoleHalfLength==0)_ShieldingHoleHalfLength=_ShieldingHalfLength;
```

Comment:
Where is _ShieldingHalfLength set before this? Is there any risk of it being ~0 but not exactly 0?
Same for line 232.